File: Palindrome Number/9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # if the number is negative it cannot be a palidrome so must return false
        if x < 0:
            return False
        

        original = x
        reverse = 0

        while original > 0:
            remainder = original % 10 
            original = original // 10

            reverse = reverse * 10 + remainder
        
        return reverse == x

# RESULTS

# Accepted
# 11511/11511 cases passed (47 ms)
# Your runtime beats 83.52 % of python3 submissions
# Your memory usage beats 40.71 % of python3 submissions (16.5 MB)
        
# @lc code=end

File: Palindrome Number/9.palindrome-number.go
/*
 * @lc app=leetcode id=9 lang=golang
 *
 * [9] Palindrome Number
 */

// @lc code=start
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	var reverse int = 0
	var original int = x

	for original > 0 {
		reverse = reverse * 10 + (original % 10)

		original = original / 10
	}

	return reverse == x

    
}

// RESULTS 
// Accepted
// 11511/11511 cases passed (12 ms)
// Your runtime beats 57.15 % of golang submissions
// Your memory usage beats 94.08 % of golang submissions (4.2 MB)
// @lc code=end

